### PR TITLE
[Master] rollback all dedicated fcp when got exception in dedicating fcp

### PR DESCRIPTION
**Pull request by dongyanyang**
With attach, there would be multiple fcp to be dedicated. Previously if
one fcp dedicate failed, only the current fcp will be rollback.
This PR will also rollback the previously dedicated fcps.

Signed-off-by: dyyang <dyyang@cn.ibm.com>

### CODE REVIEW

**Comment by dongyanyang** [comment]
image

**Comment by jichenjc** [comment]
@bjhuangr please help to take a look

File: zvmsdk/database.py [hunk 1]
```diff
@@ -433,7 +433,15 @@ def get_connections_from_fcp(self, fcp):
...
                                   "fcp_id=?", (fcp,))
-            connections = result.fetchall()
+            fcp_info = result.fetchall()
+            if not fcp_info:
+                msg = 'FCP with id: %s does not exist in DB.' % fcp
+                LOG.error(msg)
+                obj_desc = "FCP with id: %s" % fcp
+                raise exception.SDKObjectNotExistError(obj_desc=obj_desc,
+                                                       modID=self._module_id)
+            connections = fcp_info[0][0]
```
Comment (jichenjc) [inline zvmsdk/database.py:443]: I didn't go to detail about code
previously we return fetch all results, now we return [0][0] ,any problem on this?

Comment (dongyanyang) [reply zvmsdk/database.py:443]: no, there's only one call to this function before, and used "if not connections" for the check, behavior not changed.

Comment (jichenjc) [reply zvmsdk/database.py:443]: ok

**Comment by jichenjc** [comment]
lgtm

File: zvmsdk/volumeop.py [hunk 2]
```diff
@@ -641,10 +646,18 @@ def attach(self, connection_info):
...
             for i in range(path_count):
-                self._attach(fcp[i].lower(), assigner_id, target_wwpns,
-                             target_lun, multipath, os_version, mount_point,
-                             path_count, is_root_volume)
+                try:
+                    new = self._attach(fcp[i].lower(), assigner_id,
+                                       target_wwpns, target_lun, multipath,
+                                       os_version, mount_point, path_count,
+                                       is_root_volume)
+                    if new and is_root_volume is False:
```
Comment (bjhuangr) [inline zvmsdk/volumeop.py:656]: (new and is_boot) is false, or new and (is_root is flase) ? what's your expectation?

File: zvmsdk/volumeop.py [hunk 3]
```diff
@@ -641,10 +646,18 @@ def attach(self, connection_info):
...
+                        dedicated_fcp.append(fcp[i])
+                except exception.SDKBaseException:
+                    self._rollback_dedicated_fcp(dedicated_fcp, assigner_id)
```
Comment (changzhi1990) [inline zvmsdk/volumeop.py:659]: Why catch exception here?
At line 593, you have already caught the exception "exception.SDKBaseException". So here, can not catch this exception anymore. right?

File: zvmsdk/volumeop.py (hunk 2 above)
Comment (dongyanyang) [reply zvmsdk/volumeop.py:656]: new and (is_root is flase)

**Pull request merged**